File: CPlusPlus/3986.cpp
#include <bits/stdc++.h>

#define FASTIO                             \
    std::ios_base::sync_with_stdio(false); \
    std::cin.tie(0);                       \
    std::cout.tie(0)
#define ENDL '\n'

/***************************************************************************************
 *   * 링크 : https://www.acmicpc.net/problem/3986
 ***************************************************************************************/

using namespace std;

int main() {
    FASTIO;
    int n;
    cin >> n;

    int cnt = 0;
    for (int i = 0; i < n; i++) {
        string st;
        cin >> st;
        stack<char> s;

        for (int j = 0; j < st.length(); j++) {
            if (!s.empty() && s.top() == st[j]) {
                s.pop();
                continue;
            }

            s.push(st[j]);
        }

        if (s.empty()) {
            cnt++;
        }
    }

    cout << cnt << ENDL;
    return 0;
}

File: CSharp/3986.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

// 링크 : https://www.acmicpc.net/problem/3986

namespace BOJ
{
    class Program
    {
        static void Main(string[] args)
        {
            Stack<char> st = new Stack<char>();
            int answer = 0;
            int testCase = int.Parse(Console.ReadLine());

            for (int i = 0; i < testCase; i++)
            {
                string s = Console.ReadLine();

                st.Push(s[0]);

                for (int t = 1; t < s.Length; t++)
                {
                    if (st.Peek() == s[t])
                    {
                        st.Pop();
                    }
                    else
                    {
                        st.Push(s[t]);
                    }
                }

                if (st.Count == 0)
                {
                    answer++;
                }
            }

            Console.WriteLine($"{answer}");
        }
    }
}